### DISCUSSION TITLE
RFC: aggregated api-javadoc/scaladoc task

### DISCUSSION THREAD

**Discussion by Sailsman63** [com-lihaoyi/mill#1750] [Ideas]
#1688 exposes what seems to be a gap in basic services. How shall a mill build create aggregated documentation for the entire project? Per-artifact docJars are only useful when pulling the artifact into an IDE that uses them.

 - This should be a built-in, not a third-party extension
 - (Personal preference) The resulting doc should *not* be packed in a `.jar` - again, that's only for the IDE that's pulling in an artifact.
 - Not clear where this should be implemented, but at most it should only take a single `import` to make it available to the user.
 - Should there be separate tasks for JavaModules and ScalaModules, or should a single task build both? Ideally, they'd be able to cross-reference each other, but I don't know if the doc tools make that possible at all.

Thoughts and feedback welcome. I'm very new to scala, (Really only interested in mill as a build tool for my java projects) so don't be surprised if I ask silly/basic questions.

**Comment by lihaoyi** [comment]
Should be fixed by https://github.com/com-lihaoyi/mill/pull/2448, which introduces `mill.scalalib.UnidocModule`. Should land and go out in the next release
